File: server-go/internal/brain/message.go
package brain

import (
	"encoding/json"
	"log"
)

type ActionType string

const (
	NewConnection   ActionType = "new-connection"
	NotifCreateUser ActionType = "notif-create-user"
	UpdateChatRooms            = "update-chat-rooms"

	NewTextAction   = "new-text"
	JoinRoomAction  = "join-room"
	LeaveRoomAction = "leave-room"

	// TODO unk no match
	NotifNewRoom = "notif-new-room"
)

type Message struct {
	UUID           string `json:"uuid"`
	ConnectionUuid string `json:"connection_uuid"`
	UserId         UserId `json:"user_id"`
	Action         string `json:"action"`

	RoomId RoomID  `json:"room_id"`
	Sender *Client `json:"sender"`

	ChatMessage `json:"chat"`
	ChatRooms   []ChatRoomDto `json:"chatRooms"`
}

func (message *Message) encode() []byte {
	jsonData, err := json.Marshal(message)
	if err != nil {
		log.Println(err)
	}

	return jsonData
}

File: ui-vuejs/src/store/SocketConfig/message.ts
import {ChatRoomType} from 'src/store/ChatServer/model';

export enum MsgType {
  NewConnection = 'new-connection',
  NotifCreateUser = 'notif-create-user',
  UpdateChatRooms = 'update-chat-rooms',

  NewTextAction = 'new-text',
  JoinRoomAction = 'join-room',
  LeaveRoomAction = 'leave-room',

  // TODO unk no match
  RoomChange = 'room-change',
  UpdateClient = 'update-client',
}


export interface UserId {
  name: string
  uuid: string
  connection_uuid: string,
}


export interface Message {
  uuid: string,
  connection_uuid: string,
  action: MsgType,
  user_id: UserId,
}


export interface MsgUpdateRooms extends Message {
  chatRooms: ChatRoomType[]
}

export interface MsgChat extends Message {
  chat: ChatMessage
}

export interface ChatMessage {
  uuid: string,
  time: string,
  text: string[],
  sender: UserId,
  room_uuid: string,
}


export const joinRoomMsgString = (roomId: string): string => {
  const msg = {
    action: MsgType.JoinRoomAction,
    target: roomId
  }

  return JSON.stringify(msg);

}
